### ISSUE TITLE
Specifying virtualbox as --vm-driver does not work

### ISSUE THREAD

**Issue by gbraad** [minishift/minishift#434]
Starting minishift with: `minishift start --vm-driver=virtualbox` will currently fail with compiling from master branch.

```
Starting local OpenShift cluster using 'virtualbox' hypervisor...
E0220 14:11:01.893091   26749 start.go:149] Error starting the VM: Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?. Retrying.
E0220 14:11:01.894315   26749 start.go:149] Error starting the VM: Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?. Retrying.
E0220 14:11:01.894404   26749 start.go:149] Error starting the VM: Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?. Retrying.
E0220 14:11:01.894418   26749 start.go:155] Error starting the VM:  Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?
Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?
Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?
```

Reproducible on Linux, Windows. Not on masOS.

**Comment by budhrg** [comment]
This is little weird how minishift is behaving here with `--vm-driver=vbox` as supported drivers for windows are hyperv and  virtualbox as per this [src](https://github.com/minishift/minishift/blob/master/pkg/minikube/constants/constants_windows.go#L21).

I am getting error in windows for `minishift start --vm-driver=vbox` as:
```
$ minishift.exe start --vm-driver vbox
```

**Comment by budhrg** [comment]
Linux behavior on compiling minishit with latest master HEAD ()
```
➜  minishift git:(master) ./out/linux-amd64/minishift start --vm-driver virtualbox
Starting local OpenShift cluster using 'virtualbox' hypervisor...
E0220 12:10:30.583162   16490 start.go:149] Error starting the VM: Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?. Retrying.
E0220 12:10:30.583616   16490 start.go:149] Error starting the VM: Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?. Retrying.
E0220 12:10:30.583665   16490 start.go:149] Error starting the VM: Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?. Retrying.
E0220 12:10:30.583674   16490 start.go:155] Error starting the VM:  Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?
Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?
Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?

➜  minishift git:(master) ./out/linux-amd64/minishift start --vm-driver vbox      
Starting local OpenShift cluster using 'vbox' hypervisor...
F0220 12:10:36.016531   16510 cluster.go:314] Unsupported driver: vbox
```

**Comment by hferentschik** [comment]
I cannot re-produce this on OS X. `minishift start --vm-driver virtualbo` worked fine. Also we handle the driver selection ourselves:

	switch config.VMDriver {
	case "virtualbox":
		driver = createVirtualboxHost(config)
	case "vmwarefusion":
		driver = createVMwareFusionHost(config)
	case "kvm":
		driver = createKVMHost(config)
	case "xhyve":
		driver = createXhyveHost(config)
	case "hyperv":
		driver = createHypervHost(config)
	default:
		glog.Exitf("Unsupported driver: %s\n", config.VMDriver)
	}

https://github.com/minishift/minishift/blob/1ea9ac59a9dd3a7dc4c02d721cfd29a350e1daac/pkg/minikube/cluster/cluster.go#L302

I think you might be running into a variation of issue #179. Did you have an existing VM when you issues the start command using 'virtualbox' as driver. In this case the code tries to resume the existing VM and the driver paramter is pretty much ignored.

**Comment by hferentschik** [comment]
For what it's worth, I cleaned the Glide cache and rebuild master on OS X. The resulting binary still worked fine. So we might have a OS specific issue at hand.

Looking at the nature of the bug, a potential cause could be a change in docker-machine version. Looking back in history we had such a change in commit a7c68e22789913c4fa916da21b5335a26207f1fb (as part of a fix for issue #345). We there set the fixed version 0.9.0 instead of a "arbitrary" commit. Technically I don't think doing this at that stage was an absolute requirement to fix the issue, but it is the last change to the docker-machine version. After this commit we had to more dependency changes, but they were not affecting docker-machine and where in "seemingly" unrelated areas. I would be interested to know whether it makes a difference to compile a version with and w/o the above mentioned commit.

**Comment by hferentschik** [comment]
I am taking this one. I know what it is. Mea culpa. I will create a pull request.

**Comment by hferentschik** [comment]
Merged, thanks.

**Comment by LalatenduMohanty** [comment]
I can reproduce this on Fedora 25 too
```
$ minishift start --vm-driver=virtualbox
Starting local OpenShift cluster using 'virtualbox' hypervisor...
E0221 01:43:07.289804   32402 start.go:150] Error starting the VM: Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?. Retrying.
E0221 01:43:07.289969   32402 start.go:150] Error starting the VM: Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?. Retrying.
E0221 01:43:07.290022   32402 start.go:150] Error starting the VM: Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?. Retrying.
E0221 01:43:07.290030   32402 start.go:156] Error starting the VM:  Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?
Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?
Error creating new host: Driver "virtualbox" not found. Do you have the plugin binary "docker-machine" accessible in your PATH?
```

**Comment by LalatenduMohanty** [comment]
@hferentschik Somehow the mutiple commits as part of https://github.com/minishift/minishift/pull/379 is breaking the git bisect to work :(

**Comment by hferentschik** [comment]
@LalatenduMohanty no need to debug this further. I got it. Shame with beset though.

**Comment by LalatenduMohanty** [comment]
@hferentschik ack. thanks

**Comment by gbraad** [comment]
```
3f19376976fa015c797ac9ead42b61668ddc8e70	failed
ab36439c41a91553fd5bc38b43151a318dbb22de	failed (merge issues)
f2cf904b4c180d28f8771cd557c34c8f486c3d22	failed (merge issues)
9ba4c442d19eaca50438fd6ee42fc038b0b109cd	worked
78ed4a4bb31ac1d6bbd272b79c4a2983f4adeab7	worked
```

**Comment by hferentschik** [comment]
merged via pull request #441
